packages/protocol/contracts/L1/libs/LibVerifying.sol

```diff
@@ -219,13 +220,23 @@ library LibVerifying {
             fc.prover == blk.assignedProver
                 && fc.provenAt <= blk.proposedAt + blk.proofWindow
         ) {
+            uint64 proofDelay;
+            unchecked {
+                proofDelay = fc.provenAt - blk.proposedAt;
+
+                if (config.rewardMaxDelayPenalty > 0) {
+                    // Give the reward a penalty up to a small percentage.
+                    // This will encourage prover to submit proof ASAP.
+                    proofReward -= proofReward * proofDelay
+                        * config.rewardMaxDelayPenalty / 10_000 / blk.proofWindow;
+                }
```

Comment:
Should we have a larger penalty (or no reward, sent to Treasury or burnt?) if the proof was submitted after the proof window expired, or is slashing tokens enough? Because I noticed prover still gets reward for block, even if they are late, and also they get the rewardOpenMultiplier so actually it's a little bonus for being late (aside from slashing).

Reply:
Changed the code based on your suggestion. Please take another look.

Reply:
This makes more sense to me.